ui/components/component-library/text/README.mdx

```diff
@@ -0,0 +1,328 @@
+import { Story, Canvas, ArgsTable } from '@storybook/addon-docs';
+import ActionableMessage from '../../ui/actionable-message';
+import Text from './text';
+
+# Text
+
+> This Text (fka Typography) component has breaking changes in variant options and the line heights associated to each variant.
+
+Good typography improves readability, legibility and hierarchy of information.
+
+<Canvas>
+  <Story id="ui-components-component-library-text-text-stories-js--default-story" />
+</Canvas>
+
+## Props
+
+<ArgsTable of={Text} />
+
+## Usage
+
+The following describes the props and example usage for this component.
+
+### Variant
+
+Use the `variant` prop and the `TEXT` object from `./ui/helpers/constants/design-system.js` to change the font size of the component.
+
+<Canvas>
+  <Story id="ui-components-component-library-text-text-stories-js--variant" />
+</Canvas>
+
+```jsx
+// If importing from ui/components/app/[YOUR_COMPONENT]/ directory
+import Text from '../../ui/component-library/text';
+import { TEXT } from '../../../helpers/constants/design-system';
+
+
+<Text variant="DISPLAY_MD">display-md</Text>
+<Text variant="HEADING_LG">heading-lg</Text>
+<Text variant="HEADING_MD">heading-md</Text>
+<Text variant="HEADING_SM">heading-sm</Text>
+<Text variant="BODY_MD">body-md</Text>
+<Text variant="BODY_SM">body-sm</Text>
+<Text variant="BODY_XS">body-xs</Text>
+
+```
+
+### Color
+
+Use the `color` prop and the `COLOR` object from `./ui/helpers/constants/design-system.js` to change the color of the Typography component.
+
+<ActionableMessage
+  type="warning"
+  message="Do not use any colors in the DEPRECATED COLORS list. This will ensure themability and consistency across the codebase."
+/>
```

Comment:
I think we can remove this. We do have some deprecated colors but have removed all the old colors that were in the DEPRECATED COLORS list.

```suggestion
```